routers/categoriesRouter.js

```diff
@@ -0,0 +1,22 @@
+const express = require("express");
+const router = express.Router();
+
+class CategoriesRouter {
+  constructor(controller) {
+    this.controller = controller;
+  }
+
+  routes() {
+    router.get("/", this.controller.getCategories.bind(this.controller));
+    router.get("/:category", this.controller.getCategory.bind(this.controller));
+
+    router.get("/all", this.controller.getAll.bind(this.controller));
+    router.get(
+      "/all/sort/:sortBy",
+      this.controller.getSortedCategories.bind(this.controller)
+    );
```

Comment:
When request trying to use the all , it will fall to :category because :cateogry accept all string including "all". It will makes the routes /all doesn't work.